runtime/src/tokio/telemetry.rs

```diff
@@ -26,7 +33,38 @@ pub fn init(context: Context, level: Level, metrics: Option<SocketAddr>, traces:
     if let Some(cfg) = metrics {
         context
             .with_label("metrics")
-            .spawn(move |context| async move { metrics::server::serve(context, cfg).await });
+            .spawn(move |context| async move {
+                // Create a tokio listener for the metrics server.
+                //
+                // We explicitly avoid using a runtime `Listener` because
+                // it will track bandwidth used for metrics and apply a policy
+                // for read/write timeouts fit for a p2p network.
+                let listener = TcpListener::bind(cfg)
```

Comment:
This sets the precedent that the `runtime` doesn't need to use itself to provide useful capabilities (and probably shouldn't to avoid confusing the user).